### PR TITLE
nodes: add TextureView node

### PR DESCRIPTION
**Pull request by mbouron-gpsw**

### CODE REVIEW

File: tests/texture.py [hunk 1]
```diff
@@ -135,6 +155,30 @@ def _get_texture_cubemap_from_mrt_scene(cfg, samples=0):
...
+    group = ngl.Group()
+    cube = ngl.TextureCube(size=64, min_filter="linear", mag_filter="linear")
+
+    layer_base = 0
+    for layer_count, fragment in ((2, _RENDER_TO_CUBEMAP_1_FRAG), (4, _RENDER_TO_CUBEMAP_2_FRAG)):
+        program = ngl.Program(vertex=_RENDER_TO_CUBEMAP_VERT, fragment=fragment, nb_frag_output=layer_count)
+        program.update_vert_out_vars(var_uvcoord=ngl.IOVec3())
+        quad = ngl.Quad((-1, -1, 0), (2, 0, 0), (0, 2, 0))
+        render = ngl.Render(quad, program)
+        attachments = [ngl.TextureView(cube, layer) for layer in range(layer_base, layer_base + layer_count)]
```
Comment (cboesch-gpsw) [inline tests/texture.py:168]: `attachments` → `color_textures`?

File: libnodegl/node_rtt.c [hunk 2]
```diff
@@ -93,20 +93,45 @@ static const struct node_param rtt_params[] = {
     {NULL}
 };
 
+struct rtt_texture_info {
+    struct texture_priv *texture_priv;
+    int layer_base;
+    int layer_count;
+};
+
+static void get_rtt_texture_info(struct ngl_node *node, struct rtt_texture_info *info)
```
Comment (cboesch-gpsw) [inline libnodegl/node_rtt.c:102]: feel free to return the `struct  rtt_texture_info` directly

File: libnodegl/node_textureview.c [hunk 3]
```diff
@@ -41,7 +41,7 @@ static int textureview_init(struct ngl_node *node)
     struct textureview_priv *s = node->priv_data;
 
     if (node->cls->id == NGL_NODE_TEXTURE2D && s->layer) {
-        LOG(ERROR, "2d textures only have one layer");
+        LOG(ERROR, "2D textures only have one layer");
```
Comment (cboesch-gpsw) [inline libnodegl/node_textureview.c:44]: bad fixup

File: libnodegl/node_textureview.c [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+    {"layer",   NGLI_PARAM_TYPE_U32, OFFSET(layer),
+                .desc=NGLI_DOCSTRING("texture layer used for the view")},
+    {NULL}
+};
+
+static int textureview_init(struct ngl_node *node)
+{
+    struct textureview_priv *s = node->priv_data;
+
+    if (node->cls->id == NGL_NODE_TEXTURE2D && s->layer) {
```
Comment (cboesch-gpsw) [inline libnodegl/node_textureview.c:43]: wrong node: this should check `s->texture`

Comment (mbouron-gpsw) [reply libnodegl/node_textureview.c:43]: Fixed.

File: libnodegl/node_rtt.c (hunk 2 above)
Comment (mbouron-gpsw) [reply libnodegl/node_rtt.c:102]: Done.

File: tests/texture.py [hunk 5]
```diff
@@ -135,6 +155,30 @@ def _get_texture_cubemap_from_mrt_scene(cfg, samples=0):
...
 
+def _get_texture_cubemap_from_mrt_scene_2_pass(cfg, samples=0):
+    group = ngl.Group()
+    cube = ngl.TextureCube(size=64, min_filter="linear", mag_filter="linear")
+
+    layer_base = 0
+    for layer_count, fragment in ((2, _RENDER_TO_CUBEMAP_1_FRAG), (4, _RENDER_TO_CUBEMAP_2_FRAG)):
+        program = ngl.Program(vertex=_RENDER_TO_CUBEMAP_VERT, fragment=fragment, nb_frag_output=layer_count)
+        program.update_vert_out_vars(var_uvcoord=ngl.IOVec3())
+        quad = ngl.Quad((-1, -1, 0), (2, 0, 0), (0, 2, 0))
```
[the remark this answers is not in the document]
Comment (mbouron-gpsw) [reply tests/texture.py]: Done.

File: libnodegl/node_textureview.c (hunk 3 above)
Comment (mbouron-gpsw) [reply libnodegl/node_textureview.c]: Fixed.

File: tests/texture.py [hunk 6]
```diff
@@ -102,6 +102,26 @@ def texture_data_unaligned_row(cfg, h=32):
...
+void main()
+{
+    ngl_out_color[0] = vec4(0.0, 0.0, 1.0, 1.0); // top
+    ngl_out_color[1] = vec4(1.0, 1.0, 0.0, 1.0); // bottom
+    ngl_out_color[2] = vec4(0.0, 1.0, 1.0, 1.0); // back
+    ngl_out_color[3] = vec4(1.0, 0.0, 1.0, 1.0); // front
+}
+'''
+
+
```
[the remark this answers is not in the document]
Comment (mbouron-gpsw) [reply tests/texture.py:124]: Left as is for consistency.

**Review by cboesch-gpsw — APPROVED** [review]

**Pull request merged**